cmd/minishift/cmd/profile/profile_set.go

```diff
@@ -41,26 +37,47 @@ func runProfile(cmd *cobra.Command, args []string) {
 	validateArgs(args)
 	profileName := args[0]
 
+	err := unsetCurrentOcContext(profileActions.GetActiveProfile())
+	if err != nil {
+		fmt.Println(err.Error())
+	}
+
 	if !cmdUtil.IsValidProfile(profileName) {
 		err := profileActions.SetActiveProfile(profileName)
 		if err != nil {
 			atexit.ExitWithMessage(1, err.Error())
 		}
+		fmt.Printf("Profile '%s' set as active profile\n", profileName)
 	} else {
-		err := cmdUtil.SetOcContext(profileName)
+		err := profileActions.SetActiveProfile(profileName)
+		if err != nil {
+			atexit.ExitWithMessage(1, err.Error())
+		}
+		fmt.Printf("Profile '%s' set as active profile\n", profileName)
+
+		err = cmdUtil.SetOcContext(profileName)
 		if err != nil {
 			if glog.V(2) {
 				fmt.Println(fmt.Sprintf("%s", err.Error()))
 			}
-			fmt.Println(fmt.Sprintf("oc cli context could not changed for '%s'. %s", profileName, unableToSetOcContextErrorMessage))
+			fmt.Println(fmt.Sprintf("No oc CLI context is activated for profile, because the '%s' VM does not exist or is not running. Run 'minishift start' first.", profileName))
 		}
 
-		err = profileActions.SetActiveProfile(profileName)
+	}
+
+}
+
+func unsetCurrentOcContext(profileName string) error {
+	if cmdUtil.DoesVMExist(profileName) {
```

Comment:
Personally I'd take the 'if cmdUtil.DoesVMExist(profileName)' out and do the check on the caller side. This makes the code easier to follow and reveals more clearly that this unsetting of the context is conditional.

Reply:
Right  @hferentschik moved it out of `unsetCurrentOcContext` now.